csharp/src/Apache.Arrow/Arrays/ArrayDataConcatenator.cs

```diff
@@ -114,14 +107,15 @@ public void Visit(FixedSizeListType type)
             public void Visit(StructType type)
             {
                 CheckData(type, 1);
+                ArrowBuffer validityBuffer = ConcatenateValidityBuffer();
```

Comment:
Good catch.  For regression purposes, can you add a test case that would have caught this?

Reply:
The test I added for concatenating Map is what caught this. I then modified the test for concatenating struct so that it was semantically correct and the new version would have caught this too.